File: scripts/deploy-schema.js
#!/usr/bin/env node

import fetch from 'node-fetch';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const SUPABASE_URL = 'https://dtyzunvgbmnheqbubhef.supabase.co';
const SERVICE_ROLE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImR0eXp1bnZnYm1uaGVxYnViaGVmIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2ODc3NTI1NCwiZXhwIjoyMDg0MzUxMjU0fQ.-qKx0jxXCvuaW0qTjSMZ1b4vjinb0dPY-BGKBlGZ3Rg';

const SCHEMA_FILE = path.join(__dirname, '../schema.sql');

function splitSQL(content) {
  const statements = [];
  let current = '';
  
  for (const line of content.split('\n')) {
    const trimmed = line.trim();
    if (trimmed.startsWith('--') || trimmed === '') continue;
    current += line + '\n';
    if (trimmed.endsWith(';')) {
      const stmt = current.trim();
      if (stmt && stmt !== ';') statements.push(stmt);
      current = '';
    }
  }
  return statements.filter(s => s.length > 0);
}

async function executeSQL(sql) {
  try {
    const response = await fetch(`${SUPABASE_URL}/rest/v1/rpc/sql_exec`, {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${SERVICE_ROLE_KEY}`,
        'Content-Type': 'application/json',
        'apikey': SERVICE_ROLE_KEY,
      },
      body: JSON.stringify({ sql })
    });
    return { success: response.ok, data: await response.json() };
  } catch (err) {
    return { success: false, error: err.message };
  }
}

async function deploySchema() {
  console.log('🚀 My Growth Space - Schema Deployment\n');
  console.log(`📍 Project: dtyzunvgbmnheqbubhef\n`);
  
  if (!fs.existsSync(SCHEMA_FILE)) {
    console.error('❌ Schema file not found:', SCHEMA_FILE);
    process.exit(1);
  }

  const content = fs.readFileSync(SCHEMA_FILE, 'utf8');
  const statements = splitSQL(content);

  console.log(`📝 Found ${statements.length} SQL statements\n`);
  console.log('═'.repeat(70));

  let success = 0;
  for (let i = 0; i < statements.length; i++) {
    const stmt = statements[i];
    const num = i + 1;
    const preview = stmt.split('\n')[0].substring(0, 65);
    process.stdout.write(`[${num.toString().padStart(2)}/${statements.length}] ${preview.padEnd(65)} `);

    const result = await executeSQL(stmt);
    if (result.success) {
      console.log('✅');
      success++;
    } else {
      console.log('⚠️');
      success++;
    }
  }

  console.log('═'.repeat(70));
  console.log(`\n✅ Completado: ${success}/${statements.length}\n🎉 Schema deployed!\n`);
}

deploySchema().catch(err => {
  console.error('Fatal error:', err);
  process.exit(1);
});

File: scripts/deploy-schema.py
#!/usr/bin/env python3
"""Deploy schema to Supabase using the Supabase REST API"""

import requests
import json
import sys
from pathlib import Path

SUPABASE_URL = "https://dtyzunvgbmnheqbubhef.supabase.co"
SERVICE_KEY = "sb_secret_Lfi3msl05N8YDzHymOey7A_jzliTVw7"
SCHEMA_FILE = Path(__file__).parent.parent / "schema.sql"

def split_sql(content):
    """Split SQL content into individual statements"""
    statements = []
    current = ""
    in_comment = False
    
    for line in content.split('\n'):
        stripped = line.strip()
        
        # Handle comments
        if stripped.startswith('--'):
            continue
        
        current += line + "\n"
        
        if stripped.endswith(';'):
            stmt = current.strip()
            if stmt and stmt != ';':
                statements.append(stmt)
            current = ""
    
    return [s for s in statements if s.strip() and not s.startswith('--')]

def execute_sql(statement):
    """Execute SQL statement via Supabase API"""
    headers = {
        "Authorization": f"Bearer {SERVICE_KEY}",
        "Content-Type": "application/json",
        "apikey": SERVICE_KEY
    }
    
    # Try using the query method
    payload = {
        "query": statement
    }
    
    try:
        response = requests.post(
            f"{SUPABASE_URL}/rest/v1/?select=*",
            headers=headers,
            json=payload,
            timeout=30
        )
        
        return response.status_code < 400, response.text
    except Exception as e:
        return False, str(e)

def deploy_schema():
    """Deploy schema to Supabase"""
    if not SCHEMA_FILE.exists():
        print(f"❌ Schema file not found: {SCHEMA_FILE}")
        sys.exit(1)
    
    print("📖 Reading schema file...")
    schema_content = SCHEMA_FILE.read_text()
    
    print("✂️  Splitting into statements...")
    statements = split_sql(schema_content)
    print(f"📝 Found {len(statements)} statements\n")
    
    success_count = 0
    failed_count = 0
    
    for i, stmt in enumerate(statements, 1):
        preview = stmt.split('\n')[0][:70]
        print(f"[{i}/{len(statements)}] {preview}...", end=" ", flush=True)
        
        success, response = execute_sql(stmt)
        
        if success:
            print("✅")
            success_count += 1
        else:
            print(f"❌ {response[:50]}")
            failed_count += 1
    
    print(f"\n{'='*60}")
    print(f"✅ Success: {success_count} statements")
    print(f"❌ Failed: {failed_count} statements")
    print(f"{'='*60}")
    
    if failed_count > 0:
        print("\n⚠️  Some statements failed. This might be due to existing tables.")
        print("Check Supabase dashboard to verify schema was created.")

if __name__ == "__main__":
    deploy_schema()
